csfieldguide/static/interactives/cfg-equation-builder/README.md

```diff
@@ -0,0 +1,65 @@
+# CFG Equation Builder Interactive
+
+**Author:** Alasdair Smith
+
+This interactive demonstrates a context-free grammar (CFG) by allowing a user to use it to build a mathematical equation.
+
+## URL Parameters
+
+**It is recommended that parameters have all non-unreserved characters [percent-encoded](https://en.wikipedia.org/wiki/Percent-encoding).**
+
+### Basic Parameters
+
+- `productions=str`: Set the grammar productions that can be used. Lone integers, as well as strings beginning and ending with an inverted comma, are interpreted as terminals. An example of the correct syntax using the default grammar can be found below.
+- `examples=str|str|str|...`: Set the examples that can be selected by the `Next` generator option. Examples will be cycled through in the order given.
+- `hide-generator=[true|false] (default: false)`: If `true`, remove the built-in equation generator (`Simple` & `Random`).
+
+### Advanced Parameters
+
+These parameters allow more control over the built-in equation generator.
+The defaults work well for the default productions, but in most situations it is recommended that `recursion-depth` is set higher than the default and `retry-if-fail` is set to `true`.
+
+- `recursion-depth=int (default: 3)`: Set the maximum recursion depth for the built-in equation generator (`Random`). If this or `retry-if-fail` is set then the option to generate with depth 1 (`Simple`) is removed.
```

Comment:
When I set this URL parameter the option to generate random equations is gone. Is this expected? I read this as only the simple option was gone, not the random option too.